igvc_ws/src/sensors/src/imu_node.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python
+
+
+'''
+    Description: Reads/Subsrcibes imu data, process the data to determine current velocity and delta distance
+    Date Modified: 1 Feb 2019
+    Author: Ben Brenkman
+'''
+
+
+import ros_api as ros
+from ros_api import println
+import rospy
+
+import custom_msgs.msg as msgs
+
+imu_data = None # last known imu data in JSON format
+
+velocity_x = 0.0
+velocity_y = 0.0
+velocity_z = 0.0
+velocity_magnitude = 0.0
+
+distance_x = 0.0
+distance_y = 0.0
+distance_z = 0.0
+distance_magnitude = 0.0
+
+velocity_pub = None
+distance_pub = None
+
+
+'''
+    Calucates velocity in x, y, z directions and the magnitude
+    
+    args: data - data object of type msgs.imu
+'''
+def calculate_velocity(data):
+    return data.x * (data.duration / 1000.0), data.y * (data.duration / 1000.0), data.z * (data.duration / 1000.0), ((velocity_x ** 2) + (velocity_y ** 2) + (velocity_z ** 2)) ** (1/2.0)
```

Comment:
stylistically I would break each return value on it's own line since its super long

Reply:
Not sure how to do that with python